### PR TITLE
Fix seed from opt in rng sampler of multiwoz_v22

### PR DESCRIPTION
**Pull request by prajjwal1**
**Patch description**
Since the support for setting seed in `train_model.py` has been added, we can let user fix the seed in dataset sampler also.

**Testing steps**

**Other information**

### CODE REVIEW

File: parlai/tasks/multiwoz_v22/agents.py [hunk 1]
```diff
@@ -416,14 +416,16 @@ class MultiWOZv22DSTTeacher(tod_agents.TodUserSimulatorTeacher):
...
 
-    rng = np.random.RandomState(SEED)
-
     def __init__(self, opt: Opt, shared=None, *args, **kwargs):
         self.opt = opt
         self.fold = opt["datatype"].split(":")[0]
         opt["datafile"] = self.fold
         self.dpath = os.path.join(opt["datapath"], "multiwoz_v22")
         self.id = "multiwoz_v22"
+        if self.opt["seed"]:
```
Comment (moyapchen) [inline parlai/tasks/multiwoz_v22/agents.py:425]: Would recommend using `self.opt.get("seed")` to be safe, but otherwise looks fine.

**Review by moyapchen — APPROVED** [review]
See inline comment; Accepting under assumption recommendation there taken

**Comment by moyapchen** [comment]
Actually thinking about this slightly more - nothing actually adds a flag for "seed" to the argparser. How are you passing the "seed" argument down?

**Comment by prajjwal1** [comment]
@moyapchen I now perform indexing with `.get()`. Even I realized the issue you raised, I thought it would come from `train_model.py`, but seems like it has to come from `Teacher`, so the argument for `seed` is propagated from `Teacher`. Is this fine now ?

File: parlai/core/teachers.py [hunk 2]
```diff
@@ -163,6 +163,7 @@ class Teacher(Agent):
     def add_cmdline_args(
         cls, parser: ParlaiParser, partial_opt: Optional[Opt] = None
     ) -> ParlaiParser:
+        parser.add_argument('--seed', default=None)
```
Comment (klshuster) [inline parlai/core/teachers.py:166]: can we change this to `--teacher-seed`? to differentiate from the `--seed` already in train_model.py?

File: parlai/core/teachers.py [hunk 3]
```diff
@@ -163,6 +163,7 @@ class Teacher(Agent):
     def add_cmdline_args(
         cls, parser: ParlaiParser, partial_opt: Optional[Opt] = None
     ) -> ParlaiParser:
+        parser.add_argument('--teacher_seed', default=None)
```
Comment (klshuster) [inline parlai/core/teachers.py:166]: could we also enforce this to have `type=float`?

**Review by klshuster — APPROVED** [review]

**Pull request merged**